crowbar_framework/app/models/service_object.rb

```diff
@@ -1297,9 +1297,11 @@ def apply_role(role, inst, in_queue, bootstrap = false)
 
       next if bad_nodes.empty?
 
-      message = "Failed to apply the proposal to: "
+      message = "Failed to apply the proposal to "
       bad_nodes.each do |node|
-        message += "#{node}\n"
+        node_obj = NodeObject.find_node_by_name(node)
+        node_alias = node_obj.alias
```

Comment:
You can simply replace these two lines with: `node_alias = pre_cached_nodes[node].alias`, I think.